src/network/core/tcp_connect.cpp

```diff
@@ -48,6 +77,16 @@ TCPConnecter::~TCPConnecter()
 	freeaddrinfo(this->ai);
```

Comment:
This is sometimes called with NULL argument, but freeaddrinfo specs doesn't explicitly allow it.
It crashes on Haiku, and from what I can tell it will also crash on OpenBSD.

Reply:
Not completely sure why this is a comment on this PR; I guess that is also why it got unnoticed till now :) But a fix is pending in https://github.com/OpenTTD/OpenTTD/pull/9404, as that code hasn't been part of this PR for a while now :D